sdk/core/azure-core/inc/azure/core/etag.hpp

```diff
@@ -0,0 +1,194 @@
+      else
+      {
+        // Left side is weak ETag
+        if (right.IsWeak())
+        {
+          return left.m_value.GetValue().compare(right.m_value.GetValue()) == 0;
+        }
+        // Trim the weak tag
+        auto leftLength = left.m_value.GetValue().length();
+
+        if (leftLength - 2 != right.m_value.GetValue().length())
+        {
+          return false;
+        }
+        return left.m_value.GetValue().compare(2, leftLength - 2, right.m_value.GetValue()) == 0;
+      }
+      return false;
+    }
+
+  protected:
+    /**
+     * @brief Construct an empty (null) #ETag.
+     */
+    ETag() {}
+
+  public:
+    /**
+     * @brief Construct a #ETag.
+     */
+    explicit ETag(std::string etag) : m_value(std::move(etag)) {}
+
+    /**
+     * @brief Copy assignment.
+     */
+    ETag& operator=(const ETag&) = default;
+
+    /**
+     * @brief Whether @Tag is present.
+     * @return `true` if @ETag has a value, `false` otherwise.
+     */
+    const bool HasValue() { return m_value.HasValue(); }
+
+    /*
+     * @brief Return the contained string value
+     * @return #std::string
+     */
+    const std::string& ToString()
+    {
+      if (!m_value.HasValue())
+      {
+        abort();
+      }
+      return m_value.GetValue();
+    }
+
+    /**
+     * @brief Compare with \p other @ETag for equality.
+     * @param other Other @ETag to compare with.
+     * @return `true` if @ETag instances are equal, `false` otherwise.
+     */
+    bool operator==(const ETag& other) const { return ETagComparison(*this, other); }
+
+    /**
+     * @brief Compare with \p other @ETag for inequality.
+     * @param other Other @ETag to compare with.
+     * @return `true` if @ETag instances are not equal, `false` otherwise.
+     */
+    bool operator!=(const ETag& other) const { return !(*this == other); }
+
+    /**
+     * @brief Compare with \p other @ETag for inequality.
+     * @param other Other @ETag to compare with.
+     * @return `true` if @ETag instances are not equal, `false` otherwise.
+     */
+    bool IsWeak() const
+    {
+      // A null eTag is considered strong
+      if (!m_value)
+      {
+        return false;
+      }
+
+      auto& val = m_value.GetValue();
+      // Shortest valid weak etag is 4
+      //  W/""
+      if (val.length() < 4)
+      {
+        return false;
+      }
+
+      // Valid format is W/""
+      //   Must end with a " in the string
+      if (val[0] == 'W' && val[1] == '/' && val[2] == '"' && val[val.size() - 1] == '"')
+      {
+        return true;
+      }
+
+      return false;
+    }
+
+    /**
+     * @brief @ETag representing everything
+     * @notes The any ETag is *, (unquoted).  It is NOT the same as "*"
+     */
+    static const ETag& Any()
+    {
+      static ETag Any = ETag("*");
+      return Any;
+    }
+
+    /**
+     * @brief @ETag representing no @ETag present
+     */
+    static const ETag& Null()
+    {
+      static ETag Null = ETag();
+      return Null;
```

Comment:
```suggestion
      static ETag null = ETag();
      return null;
```

Reply:
I'm not sure we need this member. Do you have a scenario where it is required?

Reply:
It's for the case where a model type has an ETag field but the response had no ETag header.  Allows the model to initialize the field representing `ETag = `

Reply:
removed